distribution/xfer/download.go

```diff
@@ -62,6 +63,9 @@ type DownloadDescriptor interface {
 	// if it is unknown (for example, if it has not been downloaded
 	// before).
 	DiffID() (layer.DiffID, error)
+	// ForeignSource returns nil or the foreign layer's information and
+	// source.
+	ForeignSource() *distribution.Descriptor
```

Comment:
That might be a good invariant to enforce. The lack of digests on existing layers is an ongoing problem. I'd hate to exacerbate this problem by sending the message that only window's foreign layers have a digest.